### ISSUE TITLE
Show current date/time or last visible date/time on time axis

### ISSUE THREAD

**Issue by jimeney** [tradingview/lightweight-charts#152]
On the full Chart Library, the time at the end of the time axis is marked as per the screenshot, to give a greater idea of scale.
![image](https://user-images.githubusercontent.com/4056227/61947134-51952a80-af9c-11e9-8dcd-66aad6e15e85.png)

In the lightweight library, is it possible to set similar, so it doesn't appear like this:

![image](https://user-images.githubusercontent.com/4056227/61947213-8dc88b00-af9c-11e9-9f04-8e1b85cc702d.png)

The ideal behaviour or workaround, would be to have the option for to replicate the crosshair overlay to display permanently, perhaps with separate customisation of appearance as per this example:

![image](https://user-images.githubusercontent.com/4056227/61947307-cb2d1880-af9c-11e9-9d90-cedcc5403ebb.png)

**Comment by timocov** [comment]
> On the full Chart Library, the time at the end of the time axis is marked as per the screenshot, to give a greater idea of scale.

What you mean? They looks pretty the same for me at the moment... (and they used the same algorithm to generate that points/dates 🙂)

**Comment by jimeney** [comment]
If you look at the first screenshot, that's from the full TV library. On the daily scale, you get the month name, and the midpoint in the current month '14', you also get the current day number + 2 in the current month '28'. 

In the screenshot below it, is the same data, same timeframe, but lightweight chart library and you see the last datapoint on the time axis is '14' but no '28'.

**Comment by timocov** [comment]
Ah, I get it. Did you get data for 28th day? Or it's the future?

**Comment by jimeney** [comment]
It was in the future at that point in time, I'm assuming it printed 28 as it would have represented the date/time as the time axis meets the price axis

**Comment by timocov** [comment]
> It was in the future at that point in time

Yes, that's the difference. In lightweight-charts we don't know what time will be the next or in the future/past. The charting library (as you said "full TV library") have special very complicated mechanism (we call it "bar builder") that allows us predict/extrapolate time of the bars in the future and in the past according to the session of the symbol. We don't plan to add it to the lightweight-charts (or even open-source it).

**Comment by jimeney** [comment]
OK, understood.

Is my 'workaround' viable as a proposal instead, creating an option to force display the crosshair overlay on the time axis as an optional configuration, much like you already allow the forced display of the current price in the price axis? I don't need to see into the future, but being able to show the current candle open date/time would be perfectly adequate

**Comment by timocov** [comment]
> replicate the crosshair overlay to display permanently

For what time? For the latest one? I don't think that it's good for UX (it might confuse the people).

As another solution we can _try_ to show time mark for the latest bar (but not from crosshair one as you suggest), but I'm not sure about it either.

/cc @tradingview/lightweight-charts-maintainers

**Comment by jimeney** [comment]
This, but for the current date/time marker, would be very useful, however I will leave this to you guys to consider

![image](https://user-images.githubusercontent.com/4056227/62282842-2f8e2300-b448-11e9-8cfa-4d2b222d1c9f.png)

**Comment by bmatusiak** [comment]
This may help you,

https://jsfiddle.net/Lnqhtwj7/5/

in this fiddle ,   i created a overlay series,
```
priceLineVisible: false,
 lastValueVisible: false,
  overlay: true
```

with chart options
```
timeScale: {
    timeVisible: true,
  	secondsVisible: false,
	},
```
options are not necessary in chart but are necessary in overlaySeries


ok so i generated some data for the overlay
```
var overData = []

var time_enum = {
  "one minute": 60,
  "five minutes": 300,
  "fifteen minutes": 900,
  "one hour": 3600,
  "six hours": 21600,
  "one day": 86400
}

function resolutionTime(time, resolution) {
  return Math.floor(new Date(time) / (resolution * 1000)) * resolution;
}

var dd = resolutionTime(new Date(), time_enum["one minute"])

for (var i = 0; i < 300; i++) { 
var out = { time: dd + (time_enum["one minute"] * i), value: 0 };
  overData.push(out)
  console.log(out)
}

volumeSeries.setData(overData)
```

doing this created   invisable bars that allows the crosshair to hover over to find time scale...

**Comment by timocov** [comment]
I think that this could be done outside of lightweight-charts after #209 - you can add "empty" bars without data but with time so we'll show that time on timescale. I hope it'll partially fix the original issue.
